community/values/src/main/java/org/neo4j/values/storable/CRSCalculator.java

```diff
@@ -121,7 +128,7 @@ public double distance( PointValue p1, PointValue p2 )
             }
             else
             {
-                return distance2D;
+                throw new UnsupportedOperationException( "More than 3 dimensions are not supported for distance calculations." );
```

Comment:
Why? The code on lines 121-127 supports any number of additional dimensions. I guess the issue is that the full stack does not support more dimensions than 3, so you want to block it here too?

Reply:
Because I can't compute `avgHeight` with more than 3 dimensions. What would that even mean. That is why I decided to restrict it to 3 dimensions.

Reply:
By the way, I don't think it ever makes sense to support more than 3 dimensions for geographic. For cartesian, that is another story of course.

Reply:
The assumption I had was only the third coordinate was not parallel in geographic space, and so the merger of the first two dimensions would make the third parallel, while all higher dimensions were parallel from the beginning, and did not need to be adjusted. This is true if space is made of only three dimensions, which is true for all models I can think of.

Reply:
I do agree that this is not an important concern since we currently do not support higher dimensions full stack, and when we do we need to be very careful to define what they actually mean. The pythagoras equation used here is also based on the assumption that all dimensions are orthogonal and in the same units. That is true for normal 3D space, but may not be true for arbitrary higher dimensional models.

Reply:
I've seen 4D in geographic, but in that model the 4th dimension was not orthogonal to the 3rd, and so pythagoras would not have worked anyway. Probably best to keep this as is, and solve for the general case later.